lib/rate-limit.js

```diff
@@ -1,39 +1,47 @@
-export default function rateLimit (instance, maxRetry = 5) {
-  let attempt = 0
-  let lastErrorAt
+const attempts = {}
 
+export default function rateLimit (instance, maxRetry = 5) {
   instance.interceptors.response.use(function (response) {
     // we don't need to do anything here
     return response
   }, function (error) {
-    const {config} = error
-    if (!config || !instance.defaults.retryOnError) {
+    const {request, response, config} = error
+    if ((!request && !response) || !config || !instance.defaults.retryOnError) {
```

Comment:
Alright, I cleaned it up. Later on, when adding more logging, we might need it again FYI.